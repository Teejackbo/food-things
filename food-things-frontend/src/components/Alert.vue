<template>
  <div :class="alert.type" class="alert">
    {{ alert.message }}
  </div>
</template>

<script>
import { mapActions } from 'vuex'
import { REMOVE_ALERT_ACTION } from '@/store/modules/alerts/types'

export default {
  name: 'Alert',

  props: {
    alert: {
      value: Object,
      required: true,
    },
  },

  methods: {
    ...mapActions({
      removeAlert: REMOVE_ALERT_ACTION,
    }),
  },

  mounted() {
    setTimeout(() => {
      this.removeAlert(this.alert)
    }, this.alert.timeout || 3000)
  },
}
</script>

<style lang="scss" scoped>
@import '../style/variables';

.alert {
  background: $grey-lightest;
  border-left: 5px solid $grey-darkest;
  box-shadow: 0 2px 10px $shadow-lighter;
  margin: 1.5em 0;
  padding: 1em;
  position: relative;
  text-align: center;
  transition: all 0.5s;
  width: 100%;

  &::before {
    align-items: center;
    background: $grey-darkest;
    border-radius: 50%;
    color: $white;
    content: '?';
    display: flex;
    font-weight: 600;
    height: 20px;
    justify-content: center;
    left: -12px;
    position: absolute;
    top: -10px;
    width: 20px;
  }
}

.error {
  border-left: 5px solid $red;

  &::before {
    background: $red;
    content: '!';
  }
}

.warning {
  border-left: 5px solid $yellow;

  &::before {
    background: $yellow;
    content: '!';
  }
}

.success {
  border-left: 5px solid $primary;

  &::before {
    background: $primary;
    content: 'âœ“';
  }
}
</style>
