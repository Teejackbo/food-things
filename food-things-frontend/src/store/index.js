import Vue from 'vue'
import Vuex from 'vuex'

import user from '@/store/modules/user'
import alerts from '@/store/modules/alerts'
import food from '@/store/modules/food'

Vue.use(Vuex)

export default new Vuex.Store({
  modules: {
    user,
    alerts,
    food,
  },
  state: {},
  mutations: {},
  actions: {},
})
