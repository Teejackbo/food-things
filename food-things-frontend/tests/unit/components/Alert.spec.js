import { shallowMount } from '@vue/test-utils'
import Alert from '@/components/Alert.vue'
import $store from '@/store'

describe('Alert', () => {
  function getWrapper() {
    return shallowMount(Alert, {
      propsData: {
        alert: {
          type: 'success',
          timeout: 3000,
          message: 'This is an alert.',
        },
      },
      mocks: { $store },
    })
  }

  it('Should match snapshot with no class.', () => {
    const wrapper = shallowMount(Alert, {
      propsData: {
        alert: {
          message: 'This is an alert',
          timeout: 3000,
        },
      },
      mocks: { $store },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should match snapshot with a type of success.', () => {
    const wrapper = getWrapper()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should match snapshot with a type of warning.', () => {
    const wrapper = shallowMount(Alert, {
      propsData: {
        alert: {
          message: 'This is an alert',
          timeout: 3000,
          type: 'warning',
        },
      },
      mocks: { $store },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should match snapshot with a type of error.', () => {
    const wrapper = shallowMount(Alert, {
      propsData: {
        alert: {
          message: 'This is an alert',
          timeout: 3000,
          type: 'error',
        },
      },
      mocks: { $store },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('Renders a div.', () => {
    const wrapper = getWrapper()
    expect(wrapper.contains('div')).toBe(true)
  })

  it('Displays the alert message.', () => {
    const wrapper = getWrapper()
    expect(wrapper.text()).toBe('This is an alert.')
  })

  it('Applies the correct class.', () => {
    const wrapper = getWrapper()
    expect(wrapper.classes()).toContain('success')
  })
})
