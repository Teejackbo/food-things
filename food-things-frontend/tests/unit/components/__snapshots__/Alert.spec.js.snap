// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Should match snapshot with a type of error. 1`] = `
<div
  class="alert error"
>
  
  This is an alert

</div>
`;

exports[`Alert Should match snapshot with a type of success. 1`] = `
<div
  class="alert success"
>
  
  This is an alert.

</div>
`;

exports[`Alert Should match snapshot with a type of warning. 1`] = `
<div
  class="alert warning"
>
  
  This is an alert

</div>
`;

exports[`Alert Should match snapshot with no class. 1`] = `
<div
  class="alert"
>
  
  This is an alert

</div>
`;
